docs/src/guide/data_modeling.md

```diff
@@ -28,13 +33,79 @@ Notes:
 
 ## Using `CObject`
 
-Coming Soon. For now, see the example in [template-custom-type](https://github.com/composablesys/collabs/tree/master/template-custom-type).
+Coming Soon. For now, see the example in [template-custom-type](https://github.com/composablesys/collabs/tree/master/template-custom-type). You can also refer to the examples below.
 
 <!-- TODO: here, or ref elsewhere? (Extra guide page? Custom types? CObject typedoc?) -->
 
 ## Examples
 
-For more examples, see the [Demos](https://github.com/composablesys/collabs/tree/master/demos).
+There are some examples below. For more examples, see the [Demos](https://github.com/composablesys/collabs/tree/master/demos).
+
+### [CPairs](https://github.com/composablesys/collabs/blob/master/template-custom-type/src/custom_type.ts)
+**App:** We want to create a pair of variables.
+
+**Single-user data model:** We start by thinking a single-user pairs. 
+```ts
+class CPair<T, U>{
+  private readonly firstReg: T;
+  private readonly secondReg: U;
+
+  constructor(firstInitial: T, secondInitial: U) {
+    this.fitstReg = firstInitial;
+    this.secondReg = secondInitial;
+  }
+  get first(): T {
+    return this.firstReg;
+  }
+
+  set first(first: T) {
+    this.firstReg = first;
+  }
+
+  get second(): U {
+    return this.secondReg;
+  }
+
+  set second(second: U) {
+    this.secondReg = second;
+  }
+}
+```
+**Collaborative data model:** We now give out an example of custom data class `CPair` and hope it can record a pair collaboratively.
```

Comment:
Perhaps: "We now give a collaborative version in the form of a custom Collab `CPair`, which can hold a pair collaboratively."